### PR TITLE
Alloy experiments

### PR DESCRIPTION
**Pull request by konnov**
This PR introduces experiments with the Alloy specification of the 3SF protocol, which abstracts the features that are not important for checking `AccountableSafety`.

The last experiment `exp13` has not finished yet.

### CODE REVIEW

File: spec4d-alloy/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+The expected output is:
+
+```
+s UNSATISFIABLE
+```
+
+This means that the property `noAccountableSafety` does not have a model.
+Hence, we were not able to find a counterexample to `AccountableSafety`.
```
Comment (Kukovec) [inline spec4d-alloy/README.md:44]:
```suggestion
Hence, we were not able to find a counterexample to `AccountableSafety`, for inputs of the specified size.
```

File: spec4d-alloy/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+In addition to the above experiments, we ran a few experiments that would have
+the inputs of the size comparable to those produced by our TLA+ specification:
+
+| Input       | #blocks | #checkpoints | #signatures | #ffg_votes | #votes | runtime | memory  |
+|-------------|--------:|-------------:|------------:|-----------:|-------:|--------:|--------:|
+| [ffg-exp10] |    3    |      15      |      4      |      5     |   12   | 31 sec  | 56 MB   |
+| [ffg-exp11] |    4    |      20      |      4      |      5     |   12   | 152 sec | 94 MB   |
+| [ffg-exp12] |    5    |      25      |      4      |      5     |   12   | 234 sec | 117 MB  |
+| [ffg-exp13] |    7    |      15      |      4      |      10    |   40   | >10 days | 300 MB  |
```
Comment (Kukovec) [inline spec4d-alloy/README.md:70]: Can you add a short sentence or two after this table explaining why we think the jump from "a few seconds" to "multiple days" happens when the votes increase as such. I would mention the math we did for the smallest # of votes you need to observe finalized conflicting blocks.

**Review by Kukovec — APPROVED** [review]
LGTM

File: spec4d-alloy/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+```
+s UNSATISFIABLE
+```
+
+This means that the property `noAccountableSafety` does not have a model.
+Hence, we were not able to find a counterexample to `AccountableSafety`.
+
+## Experimental Results
+
+Similar to the experiments with a [direct SMT encoding][smt-enc] for CVC5, we
```
Comment (thpani) [inline spec4d-alloy/README.md:48]:
```suggestion
Similar to our experiments with the [direct SMT encoding][smt-enc] for CVC5, we
```

File: spec4d-alloy/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+| [ffg-exp2] |    4    |      5       |      4      |      5     |   12   | 10 sec   |  40 MB  |
+| [ffg-exp3] |    5    |      5       |      4      |      5     |   12   | 15 sec   |  45 MB  |
+| [ffg-exp4] |    3    |      6       |      4      |      6     |   15   | 57 sec   |  52 MB  |
+| [ffg-exp5] |    4    |      6       |      4      |      6     |   15   | 167 sec  |  55 MB  |
+| [ffg-exp6] |    5    |      6       |      4      |      6     |   15   | 245 sec  |  57 MB  |
+| [ffg-exp7] |    6    |      6       |      4      |      6     |   15   | 360 sec  |  82 MB  |
+| [ffg-exp8] |    5    |      7       |      4      |      6     |   24   | 1h 27m   |  156 MB |
+
+In addition to the above experiments, we ran a few experiments that would have
+the inputs of the size comparable to those produced by our TLA+ specification:
```
Comment (thpani) [inline spec4d-alloy/README.md:63]:
```suggestion
In addition to the above experiments, we ran a few experiments that have
inputs comparable in size to those produced by our TLA+ specification:
```

**Review by thpani — APPROVED** [review]
LGTM 👍 

I have not checked the individual parameterized `.als` files in detail.

**Pull request merged**